From 806a56ee2d6d24d8eb0b7464f824479cc46e6b8f Mon Sep 17 00:00:00 2001
From: Felix Lelchuk <felix.2.lelchuk@aptiv.com>
Date: Fri, 27 Jan 2023 16:56:20 +0000
Subject: [PATCH 4/4] Fix DTS for Aptiv PFE Slave

BZX-1487: Pfe slave driver is integrated on the Windriver linux

    -device tree adaptations
      -#include "s32g-pfe-slave.dtsi"
      -override pfe_slave properties for our use case:
        -unset dma-coherent as was done by OBE (=Olaf?) for the master case (why?)
        -interrupts, interrupt-names: since we only use hif1 only connect interrupts of HIF1
        -limit nxp,pfeng-hif-channels to only hif1
        -set nxp,pfeng-ihc-channel as hif1 (our HIF used for IHC)
        -set nxp,pfeng-master-channel as hif0 (HIF assigned to the master = M7)
      -disable predefined non-AUX interfaces pfesl_netif0..2:
        -by setting status = "disabled"
        -non-AUX interfaces are by design explicitly bonded to network-facing ports (EMACx) - and in our use case we want rule-based routing
      -enable the predefined AUX interface aux0sl:
        -by setting status = "okay"
        -assigning the desired MAC address as
        -local-mac-address = [ e2 51 24 01 df 37 ]
        -and assigning nxp,pfeng-hif-channels as hif1
Notes:

AUX interface shows as aux0sl under Linux (we want this one)
non-AUX interface shows as pfeNsl under Linux (should not show up as disabled on device-tree level)
it was critical to add a master AUX interface on M7 side (hif0) for IHC and to handle interrupts which drive RPC processing

Upstream-Status: Pending

Signed-off-by: Felix Lelchuk <felix.2.lelchuk@aptiv.com>
[Quanyang:
1. This patch comes from meta-aptiv/recipes-kernel/linux/files/recipes-kernel/linux/files/0001-Fix-DTS-for-Aptiv-PFE-Slave.patch
2. The commit message is from 1d1372b ("BZX-1487: Pfe slave driver is integrated on the Windriver linux"
in meta-aptiv, and I only cut out the related part.
3. Original modifications are in s32g274a-aptiv.dts and I move them to
a new file s32g274a-aptiv-pfems.dts.]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 .../dts/freescale/s32g274a-aptiv-pfems.dts    | 22 ++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/s32g274a-aptiv-pfems.dts b/arch/arm64/boot/dts/freescale/s32g274a-aptiv-pfems.dts
index 6c863a9a4d997..d87818f05240e 100644
--- a/arch/arm64/boot/dts/freescale/s32g274a-aptiv-pfems.dts
+++ b/arch/arm64/boot/dts/freescale/s32g274a-aptiv-pfems.dts
@@ -28,10 +28,30 @@ pfe_reserved_bmu2: pfebufs@83000000 {
 	};
 };
 
-&pfe_aux0 {
+&pfe_slave {
 	status = "okay";
+	/delete-property/ dma-coherent; /* Disable for cut 1.1 compatibility OBE */
+	interrupts = <GIC_SPI 191 IRQ_TYPE_EDGE_RISING>;
+	interrupt-names = "hif1";
+	nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_1>;
+	nxp,pfeng-ihc-channel = <PFE_HIF_CHANNEL_1>;
+	nxp,pfeng-master-channel = <PFE_HIF_CHANNEL_0>;
 };
 
 &pfesl_aux0 {
 	status = "okay";
+	local-mac-address = [ e2 51 24 01 df 37 ];
+	nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_1>;
+};
+
+&pfesl_netif0 {
+	status = "disabled";
+};
+
+&pfesl_netif1 {
+	status = "disabled";
+};
+
+&pfesl_netif2 {
+	status = "disabled";
 };
-- 
2.36.1

